sample_xblocks/basic/problem.py

```diff
@@ -139,7 +155,7 @@ def student_view(self, context=None):
                         }
                     }
                     var handlerUrl = runtime.handlerUrl(element, 'check')
-                    $.post(handlerUrl, JSON.stringify(data)).success(handleCheckResults);
+                    $.post(handlerUrl, JSON.stringify(data)).success(handleCheckResults(data));
```

Comment:
Yes, this appears to be an unnecessary level of indirection. `success` expects to be a passed a function reference which can simply be `handleCheckResults`, without invoking it with the parens/arg and removing the `return function(results) {` line and its corresponding closing `}`.